wgpu/src/backend/web.rs

```diff
@@ -1113,13 +1113,19 @@ impl crate::context::Context for Context {
         _adapter: &Self::AdapterId,
         _adapter_data: &Self::AdapterData,
     ) -> wgt::SurfaceCapabilities {
+        use wgt::TextureFormat::{Bgra8Unorm, Rgba16Float, Rgba8Unorm};
+        let preferred_format = self.0.get_preferred_canvas_format();
+        let mut formats = vec![Rgba8Unorm, Bgra8Unorm, Rgba16Float];
+        if let Some(index) = formats
+            .iter()
+            .position(|format| map_texture_format(*format) == preferred_format)
```

Comment:
Loop invariant code motion _should_ fix this, but let's put the map call outside of the position loop.

Reply:
Nevermind, totally misread it.